test/e2e/e2e_test.go

```diff
@@ -252,45 +293,260 @@ var _ = Describe("e2e external gateway connectivity", func() {
+			framework.Logf("Detected a HA mode KIND environment")
+			ciWorkerNodeSrc = ovnHaWorkerNode2
+			ciWorkerNodeDst = ovnHaWorkerNode3
+		}
+		By(fmt.Sprintf("Creating a container on node %s and verifying connectivity to a pod on node %s", ciWorkerNodeSrc, ciWorkerNodeDst))
+
+		// Create the pod that will be used as the destination for the connectivity test
+		createGenericPod(f, dstPingPodName, ciWorkerNodeDst, command)
+		// There is a condition somewhere with e2e WaitForPodNotPending that returns ready
+		// before calling for the IP address will succeed. This simply adds some retries.
+		for i := 1; i < getPodIPRetry; i++ {
+			pingTarget, err = getPodAddress(dstPingPodName, f.Namespace.Name)
+			if err != nil {
+				framework.Logf("Warning unable to query the test pod on node %s %v", ciWorkerNodeSrc, err)
+			}
+			if pingTarget != "<no value>" {
+				framework.Logf("Destination ping target for %s is %s", dstPingPodName, pingTarget)
+				break
+			}
+			time.Sleep(time.Second * 3)
+			framework.Logf("Retry attempt %d to get pod IP from initializing pod %s", i, dstPingPodName)
+		}
+		// Fail the test if no address is ever retrieved
+		if pingTarget == "<no value>" {
+			framework.Failf("Warning: Failed to get an IP for target pod %s, test will fail", dstPingPodName)
+		}
+		// Fail the test if the destination IP is empty
+		if pingTarget == "" {
+			framework.Failf("Warning: Failed to get an IP for target pod %s, failing test", dstPingPodName)
+		}
+		// Spin up another pod that attempts to reach the previously started pod on separate nodes
+		framework.ExpectNoError(
+			checkConnectivityPingToHost(f, ciWorkerNodeSrc, "e2e-src-ping-pod", pingTarget, ipv4PingCommand, 30))
+
+		fieldSelectorFlag := fmt.Sprintf("--field-selector=spec.nodeName=%s", ciWorkerNodeSrc)
+		kubectlOut, err := framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorFlag)
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ciWorkerNodeSrc, err)
+		}
+		ovnPodName := strings.Trim(kubectlOut, "'")
+		ovnContainerFlag := fmt.Sprintf("--container=%s", ovnContainer)
+		// dump the flowmods from br-ext to verify no counters are hit
+		kubectlOut, err = framework.RunKubectl("exec", ovnPodName, ovnNsFlag, ovnContainerFlag, "--", "ovs-ofctl", "dump-flows", "br-ext")
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnWorkerNode, err)
+		}
+		for _, flowmod := range strings.Split(kubectlOut, "\n") {
+			// filter out irrelevant lines from ofctl output
+			if strings.Contains(flowmod, "n_packets=") {
+				// verify no flowmod counters were hit in br-ext
+				if !strings.Contains(flowmod, "n_packets=0") {
+					framework.Failf("Expected packets=0 but found the flow %s", flowmod)
+				}
+			}
+		}
+	})
+})
+
+// Test e2e inter-node connectivity over br-int
+var _ = Describe("test e2e inter-node connectivity between worker nodes", func() {
```

Comment:
what's the difference between this describe block and the block on 261? They have the same description it looks like.

Reply:
Sorry about that, I got the descriptions jumbled. There are two end to end tests verifying inter-node connectivity. One is e2e with hybrid-overlay enabled and the second is without HO enabled. The test for HO also verifies no flowmods in br-ext are hit ensuring pod to pod traffic in a namespace annotated with `hybrid-overlay-external-gw` stays within `br-int`.